dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs

```diff
@@ -142,32 +141,63 @@ public override Task<Run> CreateRunAsync(string threadId, RunCreateParams runCre
                 Runs.Add(threadId, new() { newRun });
             }
 
-            return Task.FromResult(newRun);
+            return Task.FromResult(Response.FromValue(newRun, Mock.Of<Response>()));
         }
 
-        public override Task<Run> RetrieveRunAsync(string threadId, string runId, CancellationToken cancellationToken)
+        public override Task<Response<ThreadRun>> GetRunAsync(string threadId, string runId, CancellationToken cancellationToken = default)
         {
-            string nextStatus = RemainingRunStatus.Dequeue();
-            Run run = Runs[threadId].First(r => string.Equals(r.Id, runId));
-            run.Status = nextStatus;
-            return Task.FromResult(run);
+            if (Runs[threadId].Count() == 0)
+            {
+                return Task.FromResult(Response.FromValue<ThreadRun>(null!, Mock.Of<Response>()));
+            }
+
+            RunStatus runStatus = RemainingRunStatus.Dequeue();
+            int i = Runs[threadId].FindIndex(r => string.Equals(r.Id, runId));
+
+            ThreadRun run = Runs[threadId][i];
+
+            // Updates the runStatus. This cannot be done in the original run object.
+            ThreadRun runWithUpdatedStatus = AssistantsModelFactory.ThreadRun(
+                run.Id,
+                run.ThreadId,
+                run.AssistantId,
+                runStatus,
+                run.RequiredAction,
+                null,
+                run.Model,
+                run.Instructions,
+                run.Tools
+            );
+
+            Runs[threadId][i] = runWithUpdatedStatus;
+
+            return Task.FromResult(Response.FromValue(runWithUpdatedStatus, Mock.Of<Response>()));
         }
 
-        public override Task<Run?> RetrieveLastRunAsync(string threadId, CancellationToken cancellationToken)
+        public override async Task<Response<PageableList<ThreadRun>>> GetRunsAsync(string threadId, int? limit = null, ListSortOrder? order = null, string after = null, string before = null, CancellationToken cancellationToken = default)
```

Comment:
Could you handle these warnings listed below?

Reply:
fixed